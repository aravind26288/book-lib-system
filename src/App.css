.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* custom css */
body{background-color: #EFEFEF;}
.bg-dark-blue{background-color: #173c62;}
.txt-center{text-align: center;}
.txt-left{text-align: left;}
.txt-right{text-align: right;}
.book-lib-app__navbar{color: #FFF;}
.book-lib-app__navbar a, .book-lib-app__navbar a:link, .book-lib-app__navbar a:hover, .book-lib-app__navbar a:visited, .book-lib-app__navbar a:active{
  color: #FFF;
}
.book-lib-app__content-wrapper{min-height: 490px; background: #FFF; box-shadow: 1px 1px 6px 0px #173c62;}
.book-lib-app__content-title{
  border-bottom: 2px solid #173c62;  padding: 10px 0;
  font-size: 15px;  font-weight: bold;  color: #666;
}
.book-lib-app__content-wrapper input{
  border: 1px solid #173c62;  color: #173c62;
  border-radius: 20px;  padding: 10px 20px;
  height: 50px;  font-size: 14px;
}
.book-lib-app__action-btn{padding: 6px 30px; min-width: 120px;}
.book-lib-app__action-btn--submit{border: 1px solid #173c62;}
.book-lib-app__action-btn--cancel{border: 1px solid inherit;}
.book-lib-app__content-wrapper input:focus, .manage-books-form input:focus{
  box-shadow: 0 1px 3px 1px #173c62;
  transition: box-shadow 0.5s ease-in-out;
}
.manage-books-form__title{color: #173c62; font-size: 16px;}
.manage-books-form__label{font-size: 14px;}
.manage-books-form__input{
  border: 1px solid #173c62; color: #173c62;
  border-radius: 2px; padding: 10px 10px;
  height: 40px; font-size: 14px;
}
.book-lib-table th, .book-lib-table td{font-size: 14px;}
.book-lib-table th{background-color: #173c62; color: #FFF;}
.book-lib-app__book{
  border: 1px solid #d1d2d3; padding: 20px;
  margin-bottom: 10px; box-shadow: 0px 0px 1px 0px #173c62;
}
.book-lib-app__book-txtLabel{font-weight: bold;}
.book-lib-app__loading-indicator{position: absolute; top: 40%; left: 40%;}
.book-lib-app__footer{position: static; bottom: 0; background: rgba(5,5,5, 0.9); color: #FFF; padding: 10px;}
.book-lib-app__footer-content{text-align: center; font-size: 13px;}

@media (max-width: 768px){
  .book-lib-table table, thead, tbody, tr, th, td{display: block;}
  .book-lib-table thead tr{position: absolute; top: -9999px;left: -9999px;}
  td.book-lib-table-col{position: relative; padding-left: 50%;}
  td.book-lib-table-col:before{
    position: absolute; padding-right: 5px; white-space: nowrap; top: 10px;
    left: 10px; width: 45%; font-weight: bold;
  }
  td.book-lib-table-col:nth-of-type(1):before { content: "Sl.No"; }
	td.book-lib-table-col:nth-of-type(2):before { content: "Book Name"; }
	td.book-lib-table-col:nth-of-type(3):before { content: "Description"; }
	td.book-lib-table-col:nth-of-type(4):before { content: "Author"; }
}