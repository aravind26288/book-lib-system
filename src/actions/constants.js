export const bookDetails = {
    ADD_BOOK: 'ADD_BOOK',
    SEARCH_BOOK: 'SEARCH_BOOK'
}