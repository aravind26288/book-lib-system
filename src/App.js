import React from 'react';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import Home from './components/Home';
import Manage from './components/Manage';
import Layout from './components/Layout';
import './App.css';

export class App extends React.Component{
  render(){
    return (
      <Router>
        <Layout className="book-lib-app">
        <Switch>
          <Route exact path='/' component={Home} />
          <Route path='/home' component={Home} />
          <Route path='/manage' component={Manage} />
        </Switch>
        </Layout>
      </Router>
    )
  }
}

export default App;
