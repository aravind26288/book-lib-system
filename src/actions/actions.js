import {bookDetails} from './constants';

let bookId = 1;
export const addBookDetails = (book) => {
    return {
    type: bookDetails.ADD_BOOK,
    id: bookId++,
    book
    }
};

export const searchBook = (searchTerm) => ({
    type: bookDetails.SEARCH_BOOK,
    searchTerm
})