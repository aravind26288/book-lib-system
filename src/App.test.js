import React from 'React';
import App from './App';
import Enzyme, {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {addBookDetails} from './actions/actions';
import {bookDetails} from './actions/constants';
import Adapter from 'enzyme-adapter-react-16';
import {Provider} from 'react-redux';
const mockStore = configureMockStore();
Enzyme.configure({adapter: new Adapter()});
const initialAppState = {
    books:[]
};

test('manage page handles adding book details', () => {
const store = mockStore(initialAppState);
const appComp = shallow(<Provider store={store}><App /></Provider>);
 expect(appComp).toMatchSnapshot();
})