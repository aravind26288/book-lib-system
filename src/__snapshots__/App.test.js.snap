// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manage page handles adding book details 1`] = `ShallowWrapper {}`;
